packages/graph-lines/src/main.jsx

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import { GraphLines, lineUtils as utils } from '@pie-lib/charting';
+import PropTypes from 'prop-types';
+import Controls from './controls';
+import debug from 'debug';
+import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
+import { Feedback } from '@pie-lib/render-ui';
+import GraphLineControls from './GraphLineControls';
+import { withStyles } from '@material-ui/core/styles';
+
+const log = debug('pie-ui:graph-lines:main');
+
+export class Main extends React.Component {
+  static propTypes = {
+    classes: PropTypes.object,
+    session: PropTypes.shape({
+      lines: PropTypes.arrayOf(
+        PropTypes.shape({
+          from: PropTypes.shape({
+            x: PropTypes.number.isRequired,
+            y: PropTypes.number.isRequired
+          }),
+          to: PropTypes.shape({
+            x: PropTypes.number.isRequired,
+            y: PropTypes.number.isRequired
+          })
+        })
+      )
+    }),
+    onSessionChange: PropTypes.func,
+    model: PropTypes.object.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    const lines = props.session.lines || [];
+
+    this.state = {
+      session: { ...props.session, lines },
+      selection: [],
+      useSessionLines: false,
+      showCorrect: false
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This comment isn't strictly line specific. It appears to me that putting the `session` into `state` adds alot of complexity. Would it be possible to only store the selection in `state` and keep `session` in props? Then only build out your lines on `render()`. I may not be grasping fully what you're doing so let me know.

Reply:
The problem is that the user still needs to work with something on the initial loading of the components - technically the session is empty on first load. Also, the player side needs to create a copy and store it in state, because session gets cleared upon updating the config panel. What's done here is complex, but this is the implementation that takes care of the behaviour that we spoke about and agreed upon.

Reply:
As discussed, even though the logic is complex, because of the expected behaviour, it is necessary.